contracts/DNSSEC.sol

```diff
@@ -164,6 +162,41 @@ contract DNSSEC is Owned {
         emit RRSetUpdated(name);
     }
 
+    /**
+     * @dev Deletes an RR from the oracle.
+     *
+     * @param dnsclass The DNS class (1 = CLASS_INET) of the records being inserted.
+     * @param nsecname which contains the next authorative record
+     * @param deletetype The DNS record type to delete.
+     * @param deletename which you want to delete
+     * 
+     */
+    function deleteRRSet(uint16 dnsclass, bytes nsecname, uint16 deletetype, bytes deletename) public {
+        RRSet storage result = rrsets[keccak256(nsecname)][DNSTYPE_NSEC][dnsclass];
+        int compareResult = deletename.compareNames(nsecname);
+        require(compareResult >= 0);
```

Comment:
A comment here to say that the name to delete must be the same or come after the NSEC record wouldn't hurt.

Also, you can move this into the 'if' statement in the loop.